packages/clerk-js/src/ui/components/SignUp/SignUp.tsx

```diff
@@ -25,10 +25,16 @@ function SignUpRoutes(): JSX.Element {
   return (
     <Flow.Root flow='signUp'>
       <Switch>
-        <Route path='verify-email-address'>
+        <Route
+          path='verify-email-address'
+          canActivate={clerk => !!clerk.client.signUp.emailAddress}
```

Comment:
❓ Why does canActivate provide `clerk` as a first param instead of accepting a simple predicate function `() => boolean`?

Reply:
In this case, `useCoreClerk` will run under the new `RouteContext.Provider`. If we were to do it in `SignUp.tsx`, we would get different values. It might not matter, but it seemed better to me.